scalalib/src/mill/scalalib/CoursierModule.scala

```diff
@@ -104,6 +104,19 @@ trait CoursierModule extends mill.Module {
     }
   }
 
+  // bin-compat shim
+  def resolveDeps(
+      deps: Task[Agg[BoundDep]],
+      sources: Boolean,
+      artifactTypes: Option[Set[Type]]
+  ): Task[Agg[PathRef]] =
+    resolveDeps(
+      deps,
+      sources,
+      artifactTypes,
+      enableMillInternalDependencies = false
+    )
+
```

Comment:
Do you meant to open this PR against `0.12.x` branch. Otherwise the bin-compat shim isn't needed.

Reply:
> Do you meant to open this PR against `0.12.x` branch?

I wish, but I couldn't find it

Reply:
I added the shim to make the CI happy (seems it requires it), but it might not be needed, yes

Reply:
Huh, I restored the branch, but it seems to be removed again. What's going on here?

Reply:
Isn't it here? https://github.com/com-lihaoyi/mill/tree/0.12.x

Reply:
I restored it again. @lolgab removed it this time. haha.

* https://github.com/com-lihaoyi/mill/pull/4454

Reply:
I think the big green `Merge` button removes branches sometimes. I know that's what happened when I removed it on that PR, not sure whether the same thing happened to @lolgab

Reply:
I enabled the pr branch delete option in the project settings long time ago, as it is tedious to remove them manually. It's unusual to open a PR from some maintenance branch, that's why it got removed the first time and I restored it, once I recognized it. I think the second removal by @lolgab was in good intent to clean up after a merge. It's easy to overlook that the PR branch name is actually a maintenance branch. Fortunately, the restore feature worked. The `0.12.x` branch should now be there again and ready as merge target.

Reply:
I opened https://github.com/com-lihaoyi/mill/pull/4472 against the `0.12.x` branch. Waiting for it to be merged before reporting its changes here.

Reply:
I'm sorry, I just noticed the messages.
Yes, I deleted the branch as a cleanup I usually do to avoid leaving to many merged branches.
I just added `0.12.x` to Branch protection rules. I think it should help us and make it harder to delete it by mistake.
<img width="953" alt="image" src="https://github.com/user-attachments/assets/a77749e3-b3b9-4641-aa58-96ed91cb5128" />